opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchRequestBuilder.java

```diff
@@ -82,18 +88,65 @@ public OpenSearchRequestBuilder(
   /**
    * Build DSL request.
    *
-   * @return query request or scroll request
+   * @return query request with PIT or scroll request
    */
   public OpenSearchRequest build(
-      OpenSearchRequest.IndexName indexName, int maxResultWindow, TimeValue scrollTimeout) {
+      OpenSearchRequest.IndexName indexName,
+      int maxResultWindow,
+      TimeValue cursorKeepAlive,
+      OpenSearchClient client) {
+    if (this.settings.getSettingValue(Settings.Key.SQL_PAGINATION_API_SEARCH_AFTER)) {
+      return buildRequestWithPit(indexName, maxResultWindow, cursorKeepAlive, client);
+    } else {
+      return buildRequestWithScroll(indexName, maxResultWindow, cursorKeepAlive);
+    }
+  }
+
+  private OpenSearchRequest buildRequestWithPit(
```

Comment:
This method seems a copy of the original `buildRequestWithScroll`? I think the only difference is at 119 and 134 which decides to create a Scroll request or PIT request (if you added a new class)

Reply:
I've an issue created related to the improvements in branching strategy we have today for PIT and Scroll and its related implementation. This and the above comment can be addressed as part of this task: https://github.com/opensearch-project/sql/issues/2945 
Let me know if you think this is a blocking comment. If not I would like to get this addressed as part of the other task to reduce duplicate effort.